fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -145,101 +157,159 @@ private static int getFhirResourceFromBufferReader(BufferedReader resReader, int
                 if (lineRed <= numOfLinesToSkip) {
                     continue;
                 }
-                fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
-                exported++;
-                if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
-                    break;
+                try {
+                    fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
+                    exported++;
+                    if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
+                        break;
+                    }
+                } catch (FHIRParserException e) {
+                    // Log and skip the invalid FHIR resource.
+                    logger.warning("getFhirResourceFromBufferReader: " + e.getMessage());
```

Comment:
it may be worth logging out the line number as well

Reply:
thought about this also, need a little bit calculation ...

Reply:
let me try to add

Reply:
this requires to pass the processed lines of the job to this function ... let me add it ...

Reply:
it's really a little bit complicated, readitem can be run multiple time before writeitem is called, all the partition progresses are updated in writeitem, and the input stream is reused, so the passed in numofLinesToSkip is 0 for all the reads except for when it's firstly opened ...

Reply:
added, a little bit more complicated ... :)

Reply:
I think it'll be helpful.  thanks for adding